src/rgw/rgw_op.cc

```diff
@@ -4180,7 +4180,7 @@ void RGWPutObj::execute()
   }
 
   // send request to notification manager
-  const auto ret = rgw::notify::publish(s, s->object, s->obj_size, mtime, etag, rgw::notify::ObjectCreatedPut, store);
+  const auto ret = rgw::notify::publish(s, obj.key, s->obj_size, mtime, etag, rgw::notify::ObjectCreatedPut, store);
```

Comment:
for review purposes, how is obj.key improved over s->object?

Reply:
> for review purposes, how is obj.key improved over s->object?

s->object contains the object key from the request. in case of "put" it does not contain the version (only "delete" has version in the request).
obj.key contains the version in case of a versioned bucket